tests/UnifiedSpecTests/Operation.php

```diff
@@ -504,74 +651,108 @@ private function executeForTestRunner()
+                assertArrayHasKey('databaseName', $args);
+                assertArrayHasKey('collectionName', $args);
                 assertIsString($args['databaseName']);
                 assertIsString($args['collectionName']);
                 $database = $this->context->getInternalClient()->selectDatabase($args['databaseName']);
                 assertNotContains($args['collectionName'], $database->listCollectionNames());
                 break;
             case 'assertIndexExists':
+                assertArrayHasKey('databaseName', $args);
+                assertArrayHasKey('collectionName', $args);
+                assertArrayHasKey('indexName', $args);
                 assertIsString($args['databaseName']);
                 assertIsString($args['collectionName']);
                 assertIsString($args['indexName']);
                 assertContains($args['indexName'], $this->getIndexNames($args['databaseName'], $args['collectionName']));
                 break;
             case 'assertIndexNotExists':
+                assertArrayHasKey('databaseName', $args);
+                assertArrayHasKey('collectionName', $args);
+                assertArrayHasKey('indexName', $args);
                 assertIsString($args['databaseName']);
                 assertIsString($args['collectionName']);
                 assertIsString($args['indexName']);
                 assertNotContains($args['indexName'], $this->getIndexNames($args['databaseName'], $args['collectionName']));
                 break;
             case 'assertSameLsidOnLastTwoCommands':
+                /* Context::getEventObserverForClient() requires the client ID.
+                 * Avoid checking $args['client'], which is already resolved. */
+                assertArrayHasKey('client', $this->arguments);
                 $eventObserver = $this->context->getEventObserverForClient($this->arguments['client']);
                 assertEquals(...$eventObserver->getLsidsOnLastTwoCommands());
                 break;
             case 'assertDifferentLsidOnLastTwoCommands':
+                /* Context::getEventObserverForClient() requires the client ID.
+                 * Avoid checking $args['client'], which is already resolved. */
+                assertArrayHasKey('client', $this->arguments);
                 $eventObserver = $this->context->getEventObserverForClient($this->arguments['client']);
                 assertNotEquals(...$eventObserver->getLsidsOnLastTwoCommands());
                 break;
+            case 'assertNumberConnectionsCheckedOut':
+                assertArrayHasKey('connections', $args);
+                assertIsInt($args['connections']);
+                /* PHP does not implement connection pooling. Check parameters
+                 * for the sake of valid-fail tests, but otherwise raise an
+                 * error. */
+                Assert::fail('Tests using assertNumberConnectionsCheckedOut should be skipped');
```

Comment:
FWIW, this failure is no different than an earlier `connections` assertion failure. Either would satisfy the catch in `UnifiedSpecTest::testFailingTests()` (only PHPUnit's SkippedTest and Warning exceptions are disqualifying).

Still, I thought it was worth asserting as much as possible before we bomb out (on the off chance the test file itself was incorrect).